docs/CONTRIBUTING.md

```diff
@@ -27,6 +27,34 @@ Formatting rules for eclipse can be found [here](../code_style.xml).
+| all       | All levels will be logged  <br> Example: -                                                                                                                                                                                                                                                                                                 |
+| trace     | Detailed debugging (flows, request/response, details, etc.). Will have a performance impact and is therefore not for production unless it is a planned trouble shooting activity. Mainly used during development. <br>  Example: Every method call is logged in detail for a certain request and response flow. Used data is logged in detail. |
+| debug     | Simple debug logging which can be turned on and used in production if necessary (should have no impact on performance). The logs at this level should be of the type a developer might need to spot a quick fix to a problem or to at least isolate the problem further. <br> Example: Specific events with contextual details.            |
+| info      | For logging the normal flow and operation of the service(s). <br> Example: Service health, progress of requests/responses etc.                                                                                                                                                                                                             |
+| warn      | Behavior in the service(s) which are unexpected and potentially could lead to problems but were handled for the moment. However, service(s) as such are working normally and as expected. <br>Example: A primary service switching to a secondary one, connection retries, reverting to defaults etc. Should always be investigated.       |
+| error     | A service fail in the sense it cannot response to requests or data processed cannot be trusted/used.  <br>Example: Connection attempts that ultimate fail. Crucial resources not available.                                                                                                                                                                                                                                 |
+| off       | No levels will be logged <br> Example: -                                                                                                                                                                                                                                                                                                                    |                                                                                                                                                                                                                                                |
+
+This is a code example where to use isEnabled function
+ ```
+if (LOG.isDebugEnabled())
```

Comment:
Format the code example better. It is hard to read when everything is left aligned.

Reply:
You still haven't "resolved" this. Why complicate this example?